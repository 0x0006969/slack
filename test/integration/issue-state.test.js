const supertest = require('supertest');
const nock = require('nock');

const { probot, slackbot, models } = require('.');
const fixtures = require('../fixtures');

const request = supertest.agent(probot.server);

const { SlackWorkspace, SlackUser, GitHubUser } = models;

describe('Integration: change state', () => {
  let user;
  let workspace;

  beforeEach(async () => {
    process.env.EARLY_ACCESS_CHANNELS = 'C2147483705';

    user = await GitHubUser.create({
      id: 2,
      accessToken: 'github-token',
    });

    workspace = await SlackWorkspace.create({
      slackId: 'T0001',
      accessToken: 'xoxa-token',
    });

    await SlackUser.create({
      slackId: 'U2147483697',
      slackWorkspaceId: workspace.id,
      githubId: user.id,
    });
  });

  test('/github close issue', async () => {
    nock('https://api.github.com').get('/repos/owner/repo/installation')
      .reply(200, { id: 1, account: fixtures.repo.owner });
    nock('https://api.github.com')
      .patch('/repos/owner/repo/issues/123', { state: 'closed' })
      .reply(200);

    const command = fixtures.slack.command({
      text: 'close https://github.com/owner/repo/issues/123',
    });

    await request.post('/slack/command').use(slackbot).send(command)
      .expect(200, { response_type: 'in_channel' });
  });

  test('/github reopen issue', async () => {
    nock('https://api.github.com').get('/repos/owner/repo/installation')
      .reply(200, { id: 1, account: fixtures.repo.owner });
    nock('https://api.github.com')
      .patch('/repos/owner/repo/issues/123', { state: 'open' })
      .reply(200);

    const command = fixtures.slack.command({
      text: 'reopen https://github.com/owner/repo/issues/123',
    });

    await request.post('/slack/command').use(slackbot).send(command)
      .expect(200, { response_type: 'in_channel' });
  });
});
