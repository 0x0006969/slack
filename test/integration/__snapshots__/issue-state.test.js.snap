// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Integration: issue state /github close issue 1`] = `
[MockFunction cache.get] {
  "calls": Array [
    Array [
      "channel#C2147483705:issue#undefined",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`Integration: issue state /github close issue 2`] = `
[MockFunction cache.set] {
  "calls": Array [
    Array [
      "pending-command#13345224609.738474920.8088930838d88f008e0",
      Object {
        "channel_id": "C2147483705",
        "channel_name": "test",
        "command": "/github",
        "enterprise_id": "E0001",
        "enterprise_name": "Globular%20Construct%20Inc",
        "response_url": "https://hooks.slack.com/commands/1234/5678",
        "team_domain": "example",
        "team_id": "T0001",
        "text": "close https://github.com/owner/repo/issues/123",
        "token": "secret",
        "trigger_id": "13345224609.738474920.8088930838d88f008e0",
        "user_id": "U2147483697",
        "user_name": "Steve",
      },
    ],
    Array [
      "channel#C2147483705:issue#undefined",
      Object {
        "channel": "C2147483705",
        "ts": undefined,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`Integration: issue state /github close issue 3`] = `[MockFunction cache.fetch]`;

exports[`Integration: issue state /github reopen issue 1`] = `
[MockFunction cache.get] {
  "calls": Array [
    Array [
      "channel#C2147483705:issue#undefined",
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`Integration: issue state /github reopen issue 2`] = `
[MockFunction cache.set] {
  "calls": Array [
    Array [
      "pending-command#13345224609.738474920.8088930838d88f008e0",
      Object {
        "channel_id": "C2147483705",
        "channel_name": "test",
        "command": "/github",
        "enterprise_id": "E0001",
        "enterprise_name": "Globular%20Construct%20Inc",
        "response_url": "https://hooks.slack.com/commands/1234/5678",
        "team_domain": "example",
        "team_id": "T0001",
        "text": "reopen https://github.com/owner/repo/issues/123",
        "token": "secret",
        "trigger_id": "13345224609.738474920.8088930838d88f008e0",
        "user_id": "U2147483697",
        "user_name": "Steve",
      },
    ],
    Array [
      "channel#C2147483705:issue#undefined",
      Object {
        "channel": "C2147483705",
        "ts": undefined,
      },
    ],
  ],
  "results": Array [
    Object {
      "isThrow": false,
      "value": Promise {},
    },
    Object {
      "isThrow": false,
      "value": Promise {},
    },
  ],
}
`;

exports[`Integration: issue state /github reopen issue 3`] = `[MockFunction cache.fetch]`;
